cdap-app-fabric/src/main/java/co/cask/cdap/internal/profile/ProfileService.java

```diff
@@ -185,6 +186,11 @@ public void deleteProfile(ProfileId profileId) throws NotFoundException, Profile
         throw new NotFoundException(profileId);
       }
 
+      if (profileId.equals(ProfileId.NATIVE)) {
```

Comment:
this should happen outside of the transaction.

Reply:
Fixed